Content.Server/Power/EntitySystems/ExtensionCableSystem.cs

```diff
@@ -361,5 +389,40 @@ public ReceiverDisconnectedEvent(Entity<ExtensionCableReceiverComponent> receive
         }
 
         #endregion
+
+
+        private static class Distance
+        {
+            /// <summary>
+            /// Compares transform components distances to another origin transform component
+            /// </summary>
+            /// <remarks>
+            /// Does some weird XY pos comparisons when the distance is equal
+            /// </remarks>
+            public static int Compare(TransformComponent origin, TransformComponent a, TransformComponent b)
+            {
+                var distanceToA = (origin.LocalPosition - a.LocalPosition).Length();
+                var distanceToB = (origin.LocalPosition - b.LocalPosition).Length();
+                if (!MathHelper.CloseTo(distanceToA, distanceToB))
+                    return distanceToA.CompareTo(distanceToB);
+
+                // The XY check may seem unnecessary for distance comparison at first. However, it helps to make
+                // provider-receiver connections more deterministic, ruling out the spawn order from the equation.
+                if (!MathHelper.CloseTo(a.LocalPosition.Y, b.LocalPosition.Y))
+                    return a.LocalPosition.Y.CompareTo(b.LocalPosition.Y);
+                if (!MathHelper.CloseTo(a.LocalPosition.X, b.LocalPosition.X))
+                    return a.LocalPosition.X.CompareTo(b.LocalPosition.X);
+
+                return 0;
+            }
+
```

Comment:
If you want manhattan distance it should be a helper method on the engine.

Reply:
Take a closer look. It uses the Euclidean distance provided by the [Vector2.Length](https://learn.microsoft.com/en-us/dotnet/api/system.numerics.vector2.length?view=net-8.0). The X and Y shenanigans aren't needed anywhere else and shouldn't be added to the engine, I think.

Reply:
I removed it.